src/widgets/display.coffee

```diff
@@ -41,32 +41,33 @@ class Display extends Widget
   ###
   render
 
-  Replaces the older results in element with
+  Render fist page
   the given
 
   @param {Object} res
   @api public
   ###
-  render: (res) ->
-    context = extend true, res, @extraContext or {}
-    context.is_first = true
-    context.is_last = @controller.status.lastPageReached
-    @addHelpers context
-    @element.html @mustache.render @template, context
-    @trigger("df:rendered", [res])
+  render: (res) -> renderPage(res)
 
+  ###
+  renderNext
 
+  Render next Page
+  @param {Object} res
+  @api public
+  ###
+  renderNext: (res) -> renderPage(res, false)
 
   ###
-  renderNext
+  renderPage
 
   Replaces old results with the new ones in the element
   @param {Object} res
   @api public
   ###
-  renderNext: (res) ->
+  renderPage = (res, is_first = true) ->
```

Comment:
I'm going to close this, because this method in display is inherited or overwritten by the children and maybe this refactor implies take a look at them.

Besides, there's something that I'm not sure if it's ok.  In a class body, you should use the colon.
`renderPage: (res, is_first=true) ->´

Anyway, @abellosovic, we could create an issue to consider this refactor, what about it?